benches/src/utils.rs

```diff
@@ -32,9 +32,104 @@ impl ShallowTempDir {
     }
 }
 
+const DB_CLEAN_UP_ENV_VAR: &str = "DB_CLEAN_UP";
+
 impl Drop for ShallowTempDir {
     fn drop(&mut self) {
-        // Ignore errors
-        let _ = std::fs::remove_dir_all(&self.path);
+        let default_db_clean_up = true;
+        // Check if DB_CLEAN_UP is set and correctly parsed to a boolean, defaulting to true
+        let should_clean_up = env::var(DB_CLEAN_UP_ENV_VAR)
+            .map_or(default_db_clean_up, |v| {
+                v.parse::<bool>().unwrap_or(default_db_clean_up)
+            });
+
+        if should_clean_up {
+            // Ignore errors when removing the directory
+            let _ = std::fs::remove_dir_all(&self.path);
```

Comment:
they're used in benchmarks, so no, not really ~ because this is something we don't care  about *that* much if it failed/succeeded. added anyway in 978dbe0b4

Reply:
Ah right I see :+1: